modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/facet/display/context/SearchFacetTermDisplayContext.java

```diff
@@ -0,0 +1,131 @@
+ */
+
+package com.liferay.portal.search.web.internal.facet.display.context;
+
+import java.io.Serializable;
+
+import java.util.Comparator;
+
+/**
+ * @author Wade Cao
+ */
+public abstract class SearchFacetTermDisplayContext implements Serializable {
+
+	public static final Comparator<SearchFacetTermDisplayContext>
+		COMPARATOR_FREQUENCY_ASC =
+			new Comparator<SearchFacetTermDisplayContext>() {
+
+				public int compare(
+					SearchFacetTermDisplayContext
+						searchFacetTermDisplayContext1,
+					SearchFacetTermDisplayContext
+						searchFacetTermDisplayContext2) {
+
+					int result =
+						searchFacetTermDisplayContext1.getFrequency() -
+							searchFacetTermDisplayContext2.getFrequency();
+
+					if (result == 0) {
+						return _compareDisplayNames(
+							searchFacetTermDisplayContext1.getDisplayName(),
+							searchFacetTermDisplayContext2.getDisplayName());
+					}
+
+					return result;
+				}
+
+			};
+
+	public static final Comparator<SearchFacetTermDisplayContext>
+		COMPARATOR_FREQUENCY_DESC =
+			new Comparator<SearchFacetTermDisplayContext>() {
+
+				@Override
+				public int compare(
+					SearchFacetTermDisplayContext
+						searchFacetTermDisplayContext1,
+					SearchFacetTermDisplayContext
+						searchFacetTermDisplayContext2) {
+
+					int result =
+						searchFacetTermDisplayContext2.getFrequency() -
+							searchFacetTermDisplayContext1.getFrequency();
+
+					if (result == 0) {
+						return _compareDisplayNames(
+							searchFacetTermDisplayContext1.getDisplayName(),
+							searchFacetTermDisplayContext2.getDisplayName());
+					}
+
+					return result;
+				}
+
+			};
+
+	public static final Comparator<SearchFacetTermDisplayContext>
+		COMPARATOR_TERM_ASC = new Comparator<SearchFacetTermDisplayContext>() {
+
+			@Override
+			public int compare(
+				SearchFacetTermDisplayContext searchFacetTermDisplayContext1,
+				SearchFacetTermDisplayContext searchFacetTermDisplayContext2) {
+
+				int result = _compareDisplayNames(
+					searchFacetTermDisplayContext1.getDisplayName(),
+					searchFacetTermDisplayContext2.getDisplayName());
+
+				if (result == 0) {
+					return searchFacetTermDisplayContext2.getFrequency() -
+						searchFacetTermDisplayContext1.getFrequency();
+				}
+
+				return result;
+			}
+
+		};
+
+	public static final Comparator<SearchFacetTermDisplayContext>
+		COMPARATOR_TERM_DESC = new Comparator<SearchFacetTermDisplayContext>() {
+
+			@Override
+			public int compare(
+				SearchFacetTermDisplayContext searchFacetTermDisplayContext1,
+				SearchFacetTermDisplayContext searchFacetTermDisplayContext2) {
+
+				int result = _compareDisplayNames(
+					searchFacetTermDisplayContext2.getDisplayName(),
+					searchFacetTermDisplayContext1.getDisplayName());
+
+				if (result == 0) {
+					return searchFacetTermDisplayContext2.getFrequency() -
+						searchFacetTermDisplayContext1.getFrequency();
+				}
+
+				return result;
+			}
+
+		};
+
+	public abstract String getDisplayName();
```

Comment:
That idea to remove duplicates was really nice, my only concern is what should we use in Modified Facet that is using label instead of display Name?

https://github.com/liferay/liferay-portal/blob/aff2597b66b4b2453513fd44cff563288fbbd40b/modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/modified/facet/display/context/ModifiedFacetTermDisplayContext.java#L28

Reply:
@gustavolimav you may override the abstract method getDisplayName by using _label. Of course, if the variable type or usage is different, it means we have different Comparators and you can keep the Comparator in your class anyway.